### PR TITLE
First branch changes

### PR DESCRIPTION
**Pull request by Dylan-Maxwell-Wright**
added boardStars function
added gameBoardSize property to game struct
added error handling
added looping for failed input 
detailed notes in changelog

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -7,29 +7,52 @@ import (
...
+    }
     game := NewGame(boardSize)
     game.printGame()
-    row := promptForRow()
-    col := promptForCol()
-    game.placePiece(row, col, game.currentMove)
-    game.changeTurn()
-    game.printGame()
+
+    gameOn := 1
```
Comment (travis-cramer) [inline main.go:18]: haha this is a great variable name

File: game.go [hunk 2]
```diff
@@ -14,26 +14,52 @@ const (
...
         board[i] = make([]uint8, boardSize)
     }
-    newGame := Game{board: board, currentMove: DARK}
+    board = boardStars(board, boardSize)
+    newGame := Game{board: board, currentMove: DARK, gameBoardSize: boardSize}
     return newGame
 }
 
+func boardStars(board [][] uint8, boardSize int) [][] uint8 {
+    center := boardSize/2
```
Comment (travis-cramer) [inline game.go]: nit:
would `boardSize / 2` look better? (with spaces)

File: game.go [hunk 3]
```diff
@@ -14,26 +14,52 @@ const (
...
+        board [boardSize - 4][3] = 8
+        board [boardSize - 4][center] = 8
+        board [boardSize - 4][boardSize - 4] = 8
+    }
+    return board
+}
+
 func (game *Game) printGame() {
-    row := 0
+    row := 1 //edited to start at 1, so player labels start at 1, and so 0 can be used for errors in entering rows/columns
```
Comment (travis-cramer) [inline game.go]: nit:
good change, but I would remove the comment

File: main.go [hunk 4]
```diff
@@ -7,29 +7,52 @@ import (
...
     if err != nil {
         log.Fatal(err)
     }
     boardSizeAsInt, err := strconv.ParseInt(inputtedBoardSize, 10, 8)
     if err != nil {
         fmt.Println("Error parsing input. Ensure board size is a valid integer.")
-        log.Fatal(err)
+        return int(0)
+    }
+    oddTester = boardSizeAsInt % 2
```
Comment (travis-cramer) [inline main.go]: might be cleaner to put the odd tester in its own function

File: main.go [hunk 5]
```diff
@@ -7,29 +7,52 @@ import (
...
+            }
+            col := 0
+            for col == 0 {
+                col = promptForCol()
+            }
+            piecePlaced = game.placePiece(row, col, game.currentMove)
+        }
+        game.changeTurn()
+        game.printGame()
+    }
```
Comment (travis-cramer) [inline main.go:34]: all these loops are pretty fucking beautiful, so clean

File: game.go [hunk 6]
```diff
@@ -14,26 +14,52 @@ const (
...
     for i := range board {
         board[i] = make([]uint8, boardSize)
     }
-    newGame := Game{board: board, currentMove: DARK}
+    board = boardStars(board, boardSize)
+    newGame := Game{board: board, currentMove: DARK, gameBoardSize: boardSize}
     return newGame
 }
 
+func boardStars(board [][] uint8, boardSize int) [][] uint8 {
```
Comment (travis-cramer) [inline game.go]: I think this function should be like the `placePiece()` function, where it is an extension function on the `Game` model. Then it wouldn't need any extra arguments besides the game object itself.
Also, I think it should be `addBoardStars()`

**Review by travis-cramer — CHANGES_REQUESTED** [review]
left comments

File: game.go (hunk 2 above)
Comment (Dylan-Maxwell-Wright) [reply game.go]: Yeah, you're right. done

File: game.go (hunk 3 above)
Comment (Dylan-Maxwell-Wright) [reply game.go]: agreed, it was just there for you but i guess you got to see it in the pull request haha

File: main.go (hunk 4 above)
Comment (Dylan-Maxwell-Wright) [reply main.go]: Good point, I've added it just below in main.go

File: game.go (hunk 6 above)
Comment (Dylan-Maxwell-Wright) [reply game.go]: Also a really good point. Now called separately in main(), acts as a method for a game struct, and works with no arguments.

File: main.go [hunk 7]
```diff
@@ -7,29 +7,49 @@ import (
...
         log.Fatal(err)
     }
     boardSizeAsInt, err := strconv.ParseInt(inputtedBoardSize, 10, 8)
     if err != nil {
         fmt.Println("Error parsing input. Ensure board size is a valid integer.")
-        log.Fatal(err)
+        return int(0)
     }
-    return int(boardSizeAsInt)
+    findOdd = oddTester(int(boardSizeAsInt))
```
Comment (travis-cramer) [inline main.go]: here can we do
`findOdd := oddTester(int(boardSizeAsInt))`
and then remove the variable instantiation above on line 40?

File: game.go [hunk 8]
```diff
@@ -14,26 +14,50 @@ const (
 type Game struct {
     board [][]uint8
     currentMove Player
+    gameBoardSize int //added board size variable for testing out of scope moves
```
Comment (travis-cramer) [inline game.go]: nit: good change, but remove the comment

**Review by travis-cramer — APPROVED** [review]
left some comments, but mostly nits, lgtm (looks good to me)

**Pull request merged**